src/api/erc20/Erc20ApiImpl.ts

```diff
@@ -109,7 +109,7 @@ export class Erc20ApiImpl implements Erc20Api {
       from: fromAddress,
     })
 
-    if (txOptionalParams) {
+    if (txOptionalParams && txOptionalParams.onSentTransaction) {
```

Comment:
Isn't this the same as line 67 and 89? Can't you just do the same here? 

```suggestion
    if (txOptionalParams?.onSentTransaction) {
```

Reply:
missed it